### ISSUE TITLE
What is going on here?

### ISSUE THREAD

**Issue by toastal** [id3as/demo-ps#3]
I've been `purerl`-curious here for a hot minute, and am finally trying to crack into it. I enjoy PureScript, and thing the Pinto/Stetson combo looks really interesting. It was suggested that someone who wants to try just clone/use this demo repo. I just get can't get this running (well, the Nix pkgs have been downloading the last hour; update: seems the shell loaded overnight).

I know this is "a contrived purerl example", but can someone add a little more documentation on how this is running and dependencies? There's Redis, purerl, Erlang are the back-end bits?  The front-end is Halogen, Bootstrap, and jQuery? There's Docker and Nix? Can't one just use one or the other? I'm not following the back and forth between the two and the shell scripts. Things are built with Make locally or in the containers? What's Nix being used for if several of the key binaries are curl'd from Github releases into Docker (Rebar, Dhall, PureScript, Purerl, Spago)? Who is consuming this `.envrc`? Is it a Bash file or a Zsh file? 

There may not be a lot on the screen or in the db, but there's a _lot_ of stuff going on with the tooling here and it feels scattered. Something like a list of prerequisite dependencies and a [`Mermaid` diagram](https://mermaid-js.github.io/mermaid/) of the base architecture/structure in the README would go a long way.

**Comment by robashton** [comment]
@toastal sorry I didn't see this issue, I suck at keeping up  with Github notifications - you're  entirely correct that this is abhorrently documented.

A better place to try and  understand this project is to start with https://purerl-cookbook.readthedocs.io/en/latest/ and  enter the codebase  when that suggests  you do, rather than trying to look at it as a whole. As a grab bag of code examples to link to, this  project suffices, as a starting  point to learn something  - it's definitely sub par.

I've never looked at Mermaid - next time I've  got the spare cycles to give to docs/blogs/etc I'll do my best to do something with this project to make it more meaningful, and I suspect that that's a good place as any to start.

**Comment by robashton** [comment]
And goddamned yes, I really need to just punt Docker away into the bin if you're running with Nix - there is no good reason for it.

**Comment by toastal** [comment]
Nah, that's cool. I've done a bad job in the past following GitHub notifications as well because GitHub's sent to my spam folder. I know it's a lot of work to do good docs. I watched one of your talks on YouTube and was pretty interested in giving it a whirl, but ran into a roadblock when I got to this project.

**Comment by robashton** [comment]
@toastal it's a fair cop (I'm moving house at the moment so I have no bandwidth for any of this).

I should probably look at very least at collating all the info that is available into a single place, I've written a lot but it's all over the place and not structured. It's hard to do a 'fast start' because there is no such thing - the amount of scaffolding required for an Erlang project is high and doesn't get any smaller when you add another language on top of it all. Suspect I just need to bite the bullet, create a skeleton/template project on Github (as it supports such things) and build a real demo project / tutorial from that starting point. That's gonna be a good few weeks of effort though so will have to wait until I've finished unpacking :)

**Comment by robashton** [comment]
(In the meantime, there is a room set up on the fpchat Slack #purerl and all questions are answered in there pretty rapidly)
